stdlib/public/Differentiation/SIMDDifferentiation.swift.gyb

```diff
@@ -176,7 +176,7 @@ where
     )
   {
     return (lhs * rhs, { ltan, rtan in
-      return lhs * ltan + rtan * rhs
+      return lhs * rtan + ltan * rhs
```

Comment:
Do differentiation tests need to be updated too? If not, could you please add a test checking this change?

Reply:
This wasn’t caught by the tests (it seems tests only contain multiplication by a scalar, which was correct). Will add a new test

Reply:
Fixed the multiplication test (the old one did not catch the bug because it had `lhs == rhs` and `ltan == rtan`).